x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java

```diff
@@ -45,19 +45,15 @@
     private static final Logger logger = ESLoggerFactory.getLogger(LifecyclePolicy.class);
 
     public static final ParseField PHASES_FIELD = new ParseField("phases");
-    public static final ParseField TYPE_FIELD = new ParseField("type");
 
     @SuppressWarnings("unchecked")
-    public static ConstructingObjectParser<LifecyclePolicy, String> PARSER = new ConstructingObjectParser<>("lifecycle_policy", false,
-            (a, name) -> {
-                LifecycleType type = (LifecycleType) a[0];
-                List<Phase> phases = (List<Phase>) a[1];
+    public static ConstructingObjectParser<LifecyclePolicy, Tuple<String, LifecycleType>> PARSER = new ConstructingObjectParser<>("lifecycle_policy", false,
```

Comment:
I don't see why we need to have the type in the context. We are only ever parsing a timeseries lifecycle type now (hence why we are removing the type from the JSON API) so why not just hardcode the type in the call to the constructor on  Line 54?

Reply:
tests use TestLifecycleType?

Reply:
sorry, didn't see your last comment before reading this one. Sounds like we would prefer updating all the unit tests to use TimeseriesLifecycleType, so that would reduce the complexity here